deeprankcore/preprocess.py

```diff
@@ -20,30 +22,35 @@ def _preprocess_one_query(prefix: str, feature_names: List[str], query: Query):
     # because only one process may access an hdf5 file at the time:
     output_path = f"{prefix}-{os.getpid()}.hdf5"
 
-    feature_modules = [importlib.import_module(name) for name in feature_names]
+    feature_modules = [
+        importlib.import_module('deeprankcore.features.' + name) for name in feature_names]
 
     graph = query.build(feature_modules)
 
     graph.write_to_hdf5(output_path)
 
 
-def preprocess(feature_modules: List, queries: List[Query],
-               prefix: Optional[str] = None,
-               process_count: Optional[int] = None):
+def preprocess(
+    queries: List[Query],
+    prefix: Optional[str] = None,
+    process_count: Optional[int] = None,
+    feature_modules: Union[List[ModuleType], str] = "all"):
```

Comment:
I would maybe make feature_modules the second argument here. It's a core change to what happens in the function, whereas the other two arguments are less impactful.

Reply:
I'm merging the PR now, but we can discuss this further when you'll be back @DaniBodor